src/arguments/CoreBoolean.ts

```diff
@@ -0,0 +1,19 @@
+import type { PieceContext } from '@sapphire/pieces';
+import { Argument, ArgumentResult } from '../lib/structures/Argument';
+
+const truths = ['1', 'true', '+', 't', 'yes', 'y'];
+const falses = ['0', 'false', '-', 'f', 'no', 'n'];
+
+export class CoreArgument extends Argument<boolean> {
+	public constructor(context: PieceContext) {
+		super(context, { name: 'boolean' });
+	}
+
+	public run(argument: string): ArgumentResult<boolean> {
+		const boolean = argument.toLowerCase();
+		if (truths.includes(boolean)) return this.ok(true);
+		else if (falses.includes(boolean)) return this.ok(false);
```

Comment:
Useless else after `return`:
```suggestion
		if (falses.includes(boolean)) return this.ok(false);
```